spec/requests/api/v1/payments_controller_spec.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Api::V1::PaymentsController, type: :request do
+  let(:organization) { create(:organization) }
+
+  describe "POST /api/v1/payments" do
+    subject do
+      post_with_token(
+        organization,
+        "/api/v1/payments",
+        {payment: params}
+      )
+    end
+
+    let(:customer) { create(:customer, organization:) }
+    let(:invoice) { create(:invoice, organization:, customer:) }
+    let(:params) do
+      {
+        invoice_id: invoice.id,
+        amount_cents: 100,
+        reference: 'ref1'
+      }
+    end
+
+    let(:payment) { create(:payment, payable: invoice) }
+
+    before do
+      allow(ManualPayments::CreateService).to receive(:call).and_return(
+        BaseService::Result.new.tap { |r| r.payment = payment }
+      )
+    end
+
+    include_examples 'requires API permission', 'payment', 'write'
+
+    it "delegates to ManualPayments::CreateService", :aggregate_failures do
+      subject
+
+      expect(ManualPayments::CreateService).to have_received(:call).with(organization:, params:)
+
+      expect(response).to have_http_status(:success)
+      expect(json[:payment][:lago_id]).to eq(payment.id)
+      expect(json[:payment][:invoice_id]).to eq(payment.payable.id)
```

Comment:
this one is quite confusing... I have read the serializer already in main branch and the `invoice_id` could be an id if the payable_type is Invoice or a list or invoice ids if the payable type is payment request.... 😕